src/test/java/org/mifos/integrationtest/cucumber/stepdef/BillPayStepDef.java

```diff
@@ -309,4 +323,195 @@ public void iCanExtractTheCallbackBodyAndAssertTheRtpStatus() {
         }
         assertThat(flag).isTrue();
     }
+
+    @Then("I should be able to extract response body from callback for biller unidentified")
+    public void iShouldBeAbleToExtractResponseBodyFromCallbackForBillerUnidentified() {
+        boolean flag = false;
+        List<ServeEvent> allServeEvents = getAllServeEvents();
+        for (int i = allServeEvents.size() - 1; i >= 0; i--) {
+            ServeEvent request = allServeEvents.get(i);
+            if (!(request.getRequest().getBodyAsString()).isEmpty()) {
+                JsonNode rootNode = null;
+                flag = true;
+                try {
+                    rootNode = objectMapper.readTree(request.getRequest().getBody());
```

Comment:
This code will be far better if you do Json parsing to a DTO first.

Reply:
the dto will vary per tc, like for successful txns and unsuccessful txns so one single dto will not be applicable. Plus this is a common way of extracting data from request for all wiremock tcs